homeassistant/components/octoprint/binary_sensor.py

```diff
@@ -1,83 +1,98 @@
     """Set up the available OctoPrint binary sensors."""
+
     if discovery_info is None:
         return
 
     name = discovery_info["name"]
     base_url = discovery_info["base_url"]
     monitored_conditions = discovery_info["sensors"]
-    octoprint_api = hass.data[COMPONENT_DOMAIN][base_url]
+    coordinator: DataUpdateCoordinator = hass.data[COMPONENT_DOMAIN][base_url]
 
     devices = []
-    for octo_type in monitored_conditions:
-        new_sensor = OctoPrintBinarySensor(
-            octoprint_api,
-            octo_type,
-            BINARY_SENSOR_TYPES[octo_type][2],
-            name,
-            BINARY_SENSOR_TYPES[octo_type][3],
-            BINARY_SENSOR_TYPES[octo_type][0],
-            BINARY_SENSOR_TYPES[octo_type][1],
-            "flags",
-        )
-        devices.append(new_sensor)
+    if "Printing" in monitored_conditions:
+        devices.append(OctoPrintPrintingBinarySensor(coordinator, name))
+
+    if "Printing Error" in monitored_conditions:
+        devices.append(OctoPrintPrintingErrorBinarySensor(coordinator, name))
+
     add_entities(devices, True)
 
 
-class OctoPrintBinarySensor(BinarySensorEntity):
+class OctoPrintBinarySensorBase(CoordinatorEntity, BinarySensorEntity):
     """Representation an OctoPrint binary sensor."""
 
     def __init__(
-        self, api, condition, sensor_type, sensor_name, unit, endpoint, group, tool=None
+        self,
+        coordinator: DataUpdateCoordinator,
+        sensor_name: str,
+        sensor_type: str,
     ):
         """Initialize a new OctoPrint sensor."""
+        super().__init__(coordinator)
         self.sensor_name = sensor_name
-        if tool is None:
-            self._name = f"{sensor_name} {condition}"
-        else:
-            self._name = f"{sensor_name} {condition}"
+        self._name = f"{sensor_name} {sensor_type}"
         self.sensor_type = sensor_type
-        self.api = api
-        self._state = False
-        self._unit_of_measurement = unit
-        self.api_endpoint = endpoint
-        self.api_group = group
-        self.api_tool = tool
         _LOGGER.debug("Created OctoPrint binary sensor %r", self)
 
     @property
     def name(self):
         """Return the name of the sensor."""
         return self._name
 
+    @property
+    def device_class(self):
+        """Return the class of this sensor, from DEVICE_CLASSES."""
+        return None
+
     @property
     def is_on(self):
         """Return true if binary sensor is on."""
-        return bool(self._state)
+        printer = self.coordinator.data["printer"]
+        if not printer:
+            return None
 
-    @property
-    def device_class(self):
-        """Return the class of this sensor, from DEVICE_CLASSES."""
+        return bool(self._get_flag_state(printer))
+
+    def _get_flag_state(  # pylint: disable=no-self-use
+        self, printer_info: OctoprintPrinterInfo
+    ) -> bool or None:
         return None
```

Comment:
```suggestion
```

Reply:
Quick question on this one, what should I swap this out with(Apology if this is an easy question, still getting my feet under me with pyhton)? When I delete return None it complains about missing line

Reply:
There needs to be a blank line between each method.

The abstract method doesn't need a body at all.